tests/test_report.py

```diff
@@ -0,0 +1,324 @@
+        eq_(req.get_header('Content-type'), 'application/json; charset=utf-8')
+    check_request_headers_(req, base_headers, 'base')
+
+
+def get_request_url_query(req):
+    split = urllib.parse.urlsplit(req.full_url)
+    query = urllib.parse.parse_qsl(split.query)
+    if isinstance(query, list):
+        query = dict(query)
+    return query
+
+
+def check_request_query(req, query):
+    if req.method == 'POST':
+        post_data = json.loads(req.data)
+        received = post_data.get('query')
+    else:
+        query_data = get_request_url_query(req)
+        received = query_data.get('query')
+
+    if isinstance(query, bytes):
+        query = query.decode('utf-8')
+
+    query = "\n".join([s.strip() for s in query.split("\n") if s])
+    eq_(received, query)
+
+
+def check_mock_urlopen(mock_urlopen,
+                       method='POST',
+                       base_headers=None,
+                       query=None,  # defaults to `graphql_query`
+                       ):
+    assert mock_urlopen.called
+    args = mock_urlopen.call_args
+    req = args[0][0]
+    eq_(req.method, method)
+    eq_(args[1]['timeout'], timeout)
+    check_request_headers(req, base_headers)
+    check_request_query(req, query or graphql_query)
+
+
+@pytest.fixture(scope='function')
+def kms():
+    with mock_kms():
+        yield boto3.client('kms', region_name='eu-west-2')
+
+
+@httpretty.activate(allow_net_connect=False)
+@patch('urllib.request.urlopen')
+def test_report_with_valid_response(mock_urlopen, kms):
+    """Test if it generates report.csv with valid graphql response"""
+
+    with patch('reporting.report.requests.post') as mock_post:
+        set_up(kms)
+        setup_slack_api(slack_api_response_ok)
+        configure_mock_urlopen(mock_urlopen, graphql_response_ok)
+        mock_post.return_value.status_code = 200
+        mock_post.return_value.json = access_token
+        remove_csv()
+        report.handler({"emails": ["aa@gmail.com"]})
+        df = pandas.read_csv(csv_file_path)
+        assert len(df) == 1
+        assert df['ConsignmentReference'][0] == 'TDR-2022-C'
+        assert df['ConsignmentType'][0] == 'judgment'
+        assert df['TransferringBodyName'][0] == 'MOCK1 Department'
+        assert df['BodyCode'][0] == 'MOCK1'
+        assert pd.isnull(df['SeriesCode'][0]) is True
+        assert df['ConsignmentId'][0] == '71c95054-74c1-4419-8864-67046c7fbbc7'
+        assert df['UserId'][0] == '9ae3d9c5-8a71-4c50-9b19-b1ff4d315b70'
+        assert df['CreatedDateTime'][0] == '2022-05-10T11:43:19Z'
+        assert pd.isna(df['TransferInitiatedDatetime'][0]) is True
+        assert pd.isna(df['ExportDateTime'][0]) is True
+        assert pd.isna(df['ExportLocation'][0]) is True
+        assert df["FileCount"][0] == 0
+        assert df['TotalSize(Bytes)'][0] == 0
+
+
+@patch('urllib.request.urlopen')
+def test_json_error(mock_urlopen, kms):
+    """Test if broken server responses (invalid JSON) is handled"""
```

Comment:
Maybe reword test title slightly: `Test if broken server response (invalid JSON) is handled`